src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -384,4 +435,323 @@ private static String trimExcessWhitespaces(String text) {
+            final DetailNode[] child = node.getChildren();
+            if (child.length > 1
+                && child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                && getInlineTag(child) != null
+                && !ALLOWED_INLINE_TAGS.contains(getInlineTag(child).getText())) {
+                contains = false;
+                break;
+            }
+        }
+        return contains;
+    }
+
+    /**
+     * Finds and return if the HTML Format Inline Tag has allowed Inline Tags.
+     *
+     * @param ast Javadoc root Node.
+     * @return true, if first sentence contains allowed tags.
+     */
+    private static boolean validInlineTagInsideHtmlFormat(DetailNode ast) {
+        boolean contains = true;
+        if (getInlineTagNodeInsideHtml(ast) != null) {
+            for (DetailNode node : getInlineTagNodeInsideHtml(ast).getChildren()) {
+                if (node.getType() == JavadocTokenTypes.DESCRIPTION) {
+                    contains = checkInlineTagInsideDescription(node);
+                }
+            }
+        }
+        return contains;
+    }
+
+    /**
+     * Finds and return if the Inline Tag in description Node has allowed Inline Tags.
+     *
+     * @param description Description Node.
+     * @return true, if first sentence contains allowed tags.
+     */
+    private static boolean checkInlineTagInsideDescription(DetailNode description) {
+        boolean checks = true;
+        for (DetailNode nestedInlineTag : description.getChildren()) {
+            if (nestedInlineTag.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                final DetailNode[] child = nestedInlineTag.getChildren();
+                if (child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                        && !ALLOWED_INLINE_TAGS.contains(child[1].getText())) {
+                    checks = false;
+                    break;
+                }
+            }
+        }
+        return checks;
+    }
+
+    /**
+     * Get Inline tag inside Inline tag.
+     *
+     * @param javadoc Inline tag DetailNode.
+     * @return DetailNode with CUSTOM_NAME;
+     */
+    private static DetailNode getInlineTag(DetailNode... javadoc) {
+        DetailNode tagName = null;
+        if (getInlineTagNode(javadoc) != null) {
+            for (DetailNode child4 : getInlineTagNode(javadoc).getChildren()) {
+                if (child4.getType() == JavadocTokenTypes.CUSTOM_NAME) {
+                    tagName = child4;
+                }
+            }
+        }
+        return tagName;
+    }
+
+    /**
+     * Gets InlineTag Node.
+     *
+     * @param javadoc Inline tag DetailNode.
+     * @return DetailNode with CUSTOM_NAME;
+     */
+    private static DetailNode getInlineTagNode(DetailNode... javadoc) {
+        DetailNode tagNode = null;
+        for (DetailNode child2 : javadoc) {
+            if (child2.getType() == JavadocTokenTypes.DESCRIPTION) {
+                for (DetailNode child3 : child2.getChildren()) {
```

Comment:
`for (DetailNode nodeChild : node.getChildren()) {`

Reply:
Done.